examples/fungible-token/tests/workspaces.rs

```diff
@@ -0,0 +1,413 @@
+        .json::<U128>()?;
+    assert_eq!(initial_balance.0 - transfer_amount.0, root_balance.0);
+    assert_eq!(transfer_amount.0, alice_balance.0);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_close_account_empty_balance() -> anyhow::Result<()> {
+    let initial_balance = U128::from(parse_near!("10000 N"));
+    let worker = workspaces::sandbox();
+    let (contract, alice, _) = init(&worker, initial_balance).await?;
+
+    let res = alice
+        .call(&worker, contract.id().clone(), "storage_unregister")
+        .args_json((Option::<bool>::None,))?
+        .gas(300_000_000_000_000)
+        .deposit(ONE_YOCTO)
+        .transact()
+        .await?;
+    assert!(res.json::<bool>()?);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_close_account_non_empty_balance() -> anyhow::Result<()> {
+    let initial_balance = U128::from(parse_near!("10000 N"));
+    let worker = workspaces::sandbox();
+    let (contract, _, _) = init(&worker, initial_balance).await?;
+
+    let res = contract
+        .call(&worker, "storage_unregister")
+        .args_json((Option::<bool>::None,))?
+        .gas(300_000_000_000_000)
+        .deposit(ONE_YOCTO)
+        .transact()
+        .await?;
+    assert!(format!("{:?}", res.status.as_failure())
+        .contains("Can't unregister the account with the positive balance without force"));
+
+    let res = contract
+        .call(&worker, "storage_unregister")
+        .args_json((Some(false),))?
+        .gas(300_000_000_000_000)
+        .deposit(ONE_YOCTO)
+        .transact()
+        .await?;
+    assert!(format!("{:?}", res.status.as_failure())
+        .contains("Can't unregister the account with the positive balance without force"));
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn simulate_close_account_force_non_empty_balance() -> anyhow::Result<()> {
+    let initial_balance = U128::from(parse_near!("10000 N"));
+    let worker = workspaces::sandbox();
+    let (contract, _, _) = init(&worker, initial_balance).await?;
+
+    let res = contract
+        .call(&worker, "storage_unregister")
+        .args_json((Some(true),))?
+        .gas(300_000_000_000_000)
+        .deposit(ONE_YOCTO)
+        .transact()
+        .await?;
+    assert!(matches!(res.status, FinalExecutionStatus::SuccessValue(_)));
+
+    let res = contract.call(&worker, "ft_total_supply").view().await?;
+    assert_eq!(res.json::<U128>()?.0, 0);
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn simulate_transfer_call_with_burned_amount() -> anyhow::Result<()> {
+    let initial_balance = U128::from(parse_near!("10000 N"));
+    let transfer_amount = U128::from(parse_near!("100 N"));
+    let worker = workspaces::sandbox();
+    let (contract, _, defi_contract) = init(&worker, initial_balance).await?;
+
+    // defi contract must be registered as a FT account
+    register_user(&worker, &contract, defi_contract.id()).await?;
+
+    // root invests in defi by calling `ft_transfer_call`
+    // TODO: Pack into one transaction
```

Comment:
can you explain what you mean by this? Not completely intuitive

Reply:
yeah sorry for using incorrect terminology. What I meant to say is that these two actions (fn call `ft_transfer_call` and fn call `storage_unregister`) are supposed to be [batched](https://nomicon.io/RuntimeSpec/Transactions#batched-transaction) into a single transaction according to the old simtest-powered tests. Since workspaces do not support batched transactions yet, I had to split these two actions into two separate transactions. But to be fair I do not really see why you would want to batch these two specific action other than to (maybe?) save some gas... Atomicity does not matter here I think.